app/assets/javascripts/components/CartedProducts.jsx

```diff
@@ -0,0 +1,65 @@
+var CartedProducts = React.createClass({
+  getInitialState: function(){
+     return {
+       initialItems: this.props.carted_products,
+       carted_products: []
+     }
+  },
+  componentWillMount: function(){
+    this.setState({carted_products: this.state.initialItems})
+  },
+  updateQuantity: function(val, id) {
+    console.log(val, "from parent && id: ", id);
+    this.props.carted_products.map((carted_product) =>
+      (carted_product.id == id) ? (
+        carted_product.quantity = val
+      ) : (
+        carted_product
+      )
+    );
+    $.ajax({
```

Comment:
Are the drawbacks that JQuery is not light?
People are commenting that it is unnecessary to load the whole JQuery library to just make ajax calls..
Doesn't Rails already come with JQuery as well though?